src/sandstorm/supervisor.c++

```diff
@@ -644,14 +663,21 @@ kj::MainBuilder::Validity SupervisorMain::addCommandArg(kj::StringPtr arg) {
 // =====================================================================================
 
 kj::MainBuilder::Validity SupervisorMain::run() {
+  KJ_ASSERT(geteuid() != 0, "can't run as user ID zero (root)");
+  KJ_ASSERT(getegid() != 0, "can't run as group ID zero (root)");
+
   isIpTablesAvailable = checkIfIpTablesLoaded();
 
   setupSupervisor();
 
   // Exits if another supervisor is still running in this sandbox.
   systemConnector->checkIfAlreadyRunning();
 
-  SANDSTORM_LOG("Starting up grain.");
+  if (setuidMode == nullptr) {
+    SANDSTORM_LOG("Starting up grain. Sandbox type: userns");
+  } else {
+    SANDSTORM_LOG("Starting up grain. Sandbox type: setuid");
```

Comment:
As I recall, it's not literally a setuid binary that is setting up the sandbox, right? If so, the string "setuid" here is a little misleading. Maybe

> Starting up grain. Sandbox type: root

or

> Starting up grain. Sandbox type: uid-switching

Although I guess "setuid" doesn't mean "setuid root", so I _guess_ this is OK. But anyway when people read this I think they're going to think "setuid root", so I personally advise against calling it the "setuid sandbox".